software/SolarAISVN/SolarAISVN.ino

```diff
@@ -1,49 +1,65 @@
-// Solarmeter first attempt
-// Inspired by https://randomnerdtutorials.com/esp32-esp8266-publish-sensor-readings-to-google-sheets/
+// Solar- and windmeter at AISVN v0.5 2020/06/08
  
-#ifdef ESP32
-  #include <WiFi.h>
-#else
-  #include <ESP8266WiFi.h>
-#endif
-
+#include <WiFi.h>
 #include <Wire.h>
+#include <soc/sens_reg.h>
+
+RTC_DATA_ATTR int bootCount = 0;
+static RTC_NOINIT_ATTR int reg_b; // place in RTC slow memory so available after deepsleep
 
 // Replace with your SSID and Password
-const char* ssid     = "REPLACE_WITH_YOUR_SSID";
-const char* password = "REPLACE_WITH_YOUR_PASSWORD";
+const char* ssid     = "ssid";  
+const char* password = "pass";
 
 // Replace with your unique IFTTT URL resource
-const char* resource = "/trigger/solar/with/key/nAZjOphL3d-N0sne5e-h3r3-1A7gTlNSrxMJdmqy3";
+const char* resource = "/trigger/data/with/key/value";
 
 // Maker Webhooks IFTTT
 const char* server = "maker.ifttt.com";
 
 // Time to sleep
 uint64_t uS_TO_S_FACTOR = 1000000;  // Conversion factor for micro seconds to seconds
-// sleep for 30 minutes = 1800 seconds
-uint64_t TIME_TO_SLEEP = 120;
+// sleep for 2 minutes = 120 seconds
+uint64_t TIME_TO_SLEEP = 60;
+
+//    32,      33,       34,       35,   14,   26,   27,     12,   13
+// solar, battery, currentA, currentB, load, wind, dump, solar2, LiPo  
 
-int adcValue = 0;
+int voltage[9] = {0, 0, 0, 0, 0, 0, 0, 0, 0};       // all voltages in millivolt
+int pins[9] = {32, 33, 34, 35, 14, 26, 27, 12, 13};   // solar, battery, curA, curB, load, wind, dump, solar2, LiPo
+int ledPin = 5;
+int zeit = millis();
 
 void setup() {
-  Serial.begin(115200); 
-  delay(2000);
+  pinMode(ledPin, OUTPUT);
+  digitalWrite(ledPin, HIGH);  
+  Serial.begin(115200);
+  // determine cause of reset
+  esp_reset_reason_t reason = esp_reset_reason();
+  Serial.print("reason ");Serial.println(reason);
+  // get reg_b if reset not from deep sleep
+  if ((reason != ESP_RST_DEEPSLEEP)) {
+    reg_b = READ_PERI_REG(SENS_SAR_READ_CTRL2_REG);
+    Serial.println("Reading reg b.....");
+  }
+  Serial.print("reg b: ");
+  printf("%" PRIu64 "\n", reg_b);
+  delay(50);
+  digitalWrite(ledPin, LOW);
+  bootCount++;
+  delay(1000);
+  measureVoltages();
 
+  digitalWrite(ledPin, HIGH);  
   initWifi();
   makeIFTTTRequest();
-    
-  #ifdef ESP32
-    // enable timer deep sleep
-    esp_sleep_enable_timer_wakeup(TIME_TO_SLEEP * uS_TO_S_FACTOR);    
-    Serial.println("Going to sleep now");
-    // start deep sleep for 120 seconds (2 minutes)
-    esp_deep_sleep_start();
-  #else
-    // Deep sleep mode for 120 seconds (2 minutes)
-    Serial.println("Going to sleep now");
-    ESP.deepSleep(TIME_TO_SLEEP * uS_TO_S_FACTOR); 
-  #endif
```

Comment:
As mentioned above, we only use the ESP32, so the case ESP8266 needs not to be considered.